fauxfactory/__init__.py

```diff
@@ -103,15 +102,66 @@ def _unicode_letters_generator():
 UNICODE_LETTERS = [c for c in _unicode_letters_generator()]
 
 
+def _check_validation(fcn):
+    """Simple decorator to validate values generate by fcn accordingly to
+    parameters `validator`, `default` and `tries`
+
+    :param fcn: function to be enhanced
+    :return: decorated function
+    """
+
+    @wraps(fcn)
+    def validate(*args, **kwargs):
+        validator = kwargs.get('validator')
+        default = kwargs.get('default')
+        tries = kwargs.get('tries', 10)
+        if validator and default is None:
+            raise ValueError('If "validator" param is defined, "default" '
+                             'parameter must not be None')
+        if validator is None:
+            validator_fcn = lambda _: True
+        else:
+            validator_fcn=validator
+
+        if not callable(validator_fcn):
+            def regex_validator(value):
+                return re.match(validator, value)
+
+            validator_fcn = regex_validator
+
+        # Removing params related to validation but not fcn
+        for key in ('validator', 'default', 'tries'):
+            if key in kwargs:
+                kwargs.pop(key)
+
+        for _ in range(tries):
+            value = fcn(*args, **kwargs)
+            if validator_fcn(value):
+                return value
+
+        return default
+
+    return validate
+
+
 # Public Functions ------------------------------------------------------------
 
 
-def gen_string(str_type, length=None):
+def gen_string(str_type, length=None, validator=None, default=None, tries=10):
```

Comment:
Once this function only delegates to already decorated functions, decorating it would only useless apply validation 2 times.